code/noRAM/app/src/main/java/com/example/noram/OrganizerEventAttendeeActivity.java

```diff
@@ -19,5 +48,114 @@ public class OrganizerEventAttendeeActivity extends AppCompatActivity {
+            task.addOnSuccessListener(documentSnapshot -> {
+                // Get the event details
+                event = new Event();
+                event.updateWithDocument(documentSnapshot);
+
+                // Get the attendees and their check-in counts
+                event.getCheckedInAttendeesAndCounts(attendeeCallback -> {
+                    attendeeAdapter.clear();
+                    attendeeAdapter.addAll(attendeeCallback);
+                    attendeeAdapter.notifyDataSetChanged();
+                    findViewById(R.id.organizer_event_attendee_loading).setVisibility(View.GONE);
+                });
+            });
+
+        } else {
+            throw new RuntimeException("Must pass Event with key \"event\" using intent.putExtra(\"event\", eventIDAsString);");
+        }
+
+        // connect searchbar to listen for user input
+        searchInput.addTextChangedListener(new TextWatcher() {
+            /**
+             * This method is called to notify you that, somewhere within s, the text has been changed.
+             * @param s The text that has been changed
+             * @param start The starting index of the changed part in the text
+             * @param after The length of the changed part in the s sequence since the start index
+             * @param count The length of the new sequence in s
+             */
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                // do nothing
+            }
+
+            /**
+             * This method is called to notify you that, somewhere within s, the text has been changed.
+             * @param s The text that has been changed
+             * @param start The starting index of the changed part in the text
+             * @param before The length of the changed part in the s sequence since the start index
+             * @param count The length of the new sequence in s
+             */
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                // do nothing
+            }
+
+            /**
+             * This method is called to notify you that, somewhere within s, the text has been changed.
+             * @param editable The text that has been changed
+             */
+            @Override
+            public void afterTextChanged(Editable editable) {
+                searchAttendees(editable.toString().toLowerCase(Locale.ROOT));
+            }
+        });
+
+        // Set up the back button
+        findViewById(R.id.organizer_event_attendee_back).setOnClickListener(v -> finish());
+    }
+
+    /**
+     * Show search results on the searchList if the search is non-empty. Otherwise, show all events.
+     * @param search The input of the user in the search, used in the database's query
+     */
+    public void searchAttendees(String search) {
```

Comment:
i think we talked a bit about this, but we might want to pull search logic out here. we're doing searching in a lot of places and might be able to re-use some of the logic - also lets us unit test. not sure if you think this is feasible but seems worth discussing

Reply:
It would be good to pull out the logic and make it consistent, I'm searching a bit differently than in other places, but it is definitely something we should consider.